### PR TITLE
bibtex: display information from linked records in bibtex format

### PR DESCRIPTION
**Pull request by miguelgrc**
* INSPIR-3121

### CODE REVIEW

File: backend/inspirehep/records/marshmallow/literature/bibtex.py [hunk 1]
```diff
@@ -246,9 +248,23 @@ def get_series(self, data):
...
-        book_series_title = get_value(data, "book_series.title[0]")
-        conference_record_title = get_value(data, "titles.title[0]")
-        return book_series_title or conference_record_title
+        try:
+            book_records = InspireRecord.get_linked_records_from_dict_field(
+                data, "publication_info.parent_record"
+            )
+            book_record = next(book_records, None)
+            return get_value(book_record, "titles.title[0]")
+        except TypeError:
```
Comment (michamos) [inline backend/inspirehep/records/marshmallow/literature/bibtex.py]: Is this to guard against `book_record` being `None`? You can fix this by doing `book_record = next(book_record, {})` instead.

File: backend/inspirehep/records/marshmallow/literature/bibtex.py [hunk 2]
```diff
@@ -246,9 +248,23 @@ def get_series(self, data):
...
+            pass
+
+        try:
+            conference_records = InspireRecord.get_linked_records_from_dict_field(
+                data, "publication_info.conference_record"
+            )
+            conference_record = next(conference_records, None)
+            return get_value(conference_record, "titles.title[0]")
+        except TypeError:
```
Comment (michamos) [inline backend/inspirehep/records/marshmallow/literature/bibtex.py]: same idea

File: backend/inspirehep/records/marshmallow/literature/bibtex.py [hunk 3]
```diff
@@ -246,9 +248,23 @@ def get_series(self, data):
...
     def get_book_title(self, data):
-        book_series_title = get_value(data, "book_series.title[0]")
-        conference_record_title = get_value(data, "titles.title[0]")
-        return book_series_title or conference_record_title
+        try:
+            book_records = InspireRecord.get_linked_records_from_dict_field(
+                data, "publication_info.parent_record"
+            )
+            book_record = next(book_records, None)
+            return get_value(book_record, "titles.title[0]")
```
Comment (michamos) [inline backend/inspirehep/records/marshmallow/literature/bibtex.py]:
```suggestion
            return self.get_title(book_record)
```

File: backend/inspirehep/records/marshmallow/literature/bibtex.py [hunk 4]
```diff
@@ -246,9 +248,23 @@ def get_series(self, data):
...
+        except TypeError:
+            pass
+
+        try:
+            conference_records = InspireRecord.get_linked_records_from_dict_field(
+                data, "publication_info.conference_record"
+            )
+            conference_record = next(conference_records, None)
+            return get_value(conference_record, "titles.title[0]")
```
Comment (michamos) [inline backend/inspirehep/records/marshmallow/literature/bibtex.py]:
```suggestion
            return self.get_title(conference_record)
```

File: backend/inspirehep/records/marshmallow/literature/bibtex.py (hunk 1 above)
Comment (miguelgrc) [reply backend/inspirehep/records/marshmallow/literature/bibtex.py]: Also `get_linked_records_from_dict_field` returns an empty list when there's no linked record, which throws an error when calling `next` on it. With the try-except I can also avoid checking if it returns a generator or an empty list.

**Pull request merged**